txpipe/test/test_cal.py

```diff
@@ -0,0 +1,79 @@
+from ..utils.calibration_tools import MeanShearInBins
+import numpy as np
+
+def test_mean_shear():
+    name = "x"
+    limits = [-1., 0., 1.]
+    delta_gamma = 0.02
+
+    # equal weights
+    b1 = MeanShearInBins(name, limits, delta_gamma)
+
+    data = {
+        "x": np.array([-0.5, -0.5, -0.5, -0.5, 0.5, 0.5, 0.5, 0.5]),
+        "x_1p": np.array([-0.5, -0.5, -0.5, -0.5, 0.5, 0.5, 0.5, 0.5]),
+        "x_1m": np.array([-0.5, -0.5, -0.5, -0.5, 0.5, 0.5, 0.5, 0.5]),
+        "x_2p": np.array([-0.5, -0.5, -0.5, -0.5, 0.5, 0.5, 0.5, 0.5]),
+        "x_2m": np.array([-0.5, -0.5, -0.5, -0.5, 0.5, 0.5, 0.5, 0.5]),
+        "mcal_g1": np.array([-0.7, -0.6, -0.4, -0.3, 0.7, 0.6, 0.4, 0.3]),
+        "mcal_g1_1p": np.array([-0.7, -0.6, -0.4, -0.3, 0.7, 0.6, 0.4, 0.3]) + 0.5*delta_gamma,
+        "mcal_g1_1m": np.array([-0.7, -0.6, -0.4, -0.3, 0.7, 0.6, 0.4, 0.3]) - 0.5*delta_gamma,
+        "mcal_g1_2p": np.array([-0.7, -0.6, -0.4, -0.3, 0.7, 0.6, 0.4, 0.3]),
+        "mcal_g1_2m": np.array([-0.7, -0.6, -0.4, -0.3, 0.7, 0.6, 0.4, 0.3]),
+        "mcal_g2": 2*np.array([-0.7, -0.6, -0.4, -0.3, 0.7, 0.6, 0.4, 0.3]),
+        "mcal_g2_1p": 2*np.array([-0.7, -0.6, -0.4, -0.3, 0.7, 0.6, 0.4, 0.3]),
+        "mcal_g2_1m": 2*np.array([-0.7, -0.6, -0.4, -0.3, 0.7, 0.6, 0.4, 0.3]),
+        "mcal_g2_2p": 2*np.array([-0.7, -0.6, -0.4, -0.3, 0.7, 0.6, 0.4, 0.3]) + 0.5*delta_gamma,
+        "mcal_g2_2m": 2*np.array([-0.7, -0.6, -0.4, -0.3, 0.7, 0.6, 0.4, 0.3]) - 0.5*delta_gamma,
+        "weight": np.array([1, 1, 1, 1, 1, 1, 1, 1]),
+    }
+    b1.add_data(data)
+
+    mu, g1, g2, sigma1, sigma2 = b1.collect()
+    assert np.allclose(mu, [-0.5, 0.5])
+    assert np.allclose(g1, [-0.5, 0.5])
+    assert np.allclose(g2, [-1.0, 1.0])
+    expected_sigma1 = np.std([-0.2, -0.1, 0.1, 0.2]) / np.sqrt(4)
```

Comment:
Hey Joe, taking a look at this. Is this meant to be one value or std in the [-1., 0., 1.] bins? Where is [-0.2, -0.1, 0.1, 0.2] coming from?

Reply:
To make the test slightly less trivial I was using `var([0.7, 0.6, 0.4, 0.3]) == var([-0.2, -0.1, 0.1, 0.2])`, since we just shift all the numbers off.  The std dev should be the same for the two pixels as one is just the negative of the other.  I've added some explanatory comments.